docs/aws/README.md

```diff
@@ -9,102 +9,42 @@ CLUSTER_NAME=<your_cluster>
 REGION=<your_region>
 ```
 
-### Create the Karpenter IAM Policy
-This command will create an IAM Policy with access to all of the resources for all of Karpenter's features. For increased security, you may wish to reduce the permissions according to your use case.
+### Create Karpenter IAM Resources
+This command will create IAM resources used by Karpenter. For production use, please review and restrict these permissions as necessary.
 ```
-aws iam create-policy --policy-name Karpenter --policy-document "$(cat <<-EOM
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Action": [
-        "eks:DescribeNodegroup",
-        "eks:UpdateNodegroupConfig"
-      ],
-      "Effect": "Allow",
-      "Resource": "*"
-    },
-    {
-      "Action": [
-        "autoscaling:DescribeAutoScalingGroups",
-        "autoscaling:UpdateAutoScalingGroup"
-      ],
-      "Effect": "Allow",
-      "Resource": "*"
-    },
-    {
-      "Action": [
-        "sqs:GetQueueAttributes",
-        "sqs:GetQueueUrl"
-      ],
-      "Effect": "Allow",
-      "Resource": "*"
-    },
-    {
-      "Action": [
-        "ec2:DescribeLaunchTemplates",
-        "ec2:CreateLaunchTemplate",
-        "ec2:CreateFleet",
-        "ec2:RunInstances",
-        "ec2:DescribeInstances",
-        "ec2:CreateTags",
-        "ec2:DescribeSubnets",
-        "eks:DescribeCluster",
-        "iam:GetRole",
-        "iam:CreateRole",
-        "iam:AddRoleToInstanceProfile",
-        "iam:PassRole",
-        "iam:GetInstanceProfile",
-        "iam:CreateInstanceProfile",
-        "iam:AttachRolePolicy"
-      ],
-      "Effect": "Allow",
-      "Resource": "*"
-    }
-  ]
-}
-EOM
-)"
+// TODO, point to github raw uri
+aws cloudformation deploy \
+  --stack-name Karpenter \
+  --template-file ./docs/aws/karpenter.cloudformation.yaml \
```

Comment:
Do we want to have this yaml file in config? We have all other yaml files under config currently.

Reply:
When I tried to run this command, I get the following output - 

```
aws cloudformation deploy \
  --stack-name Karpenter \
  --template-file ./docs/aws/karpenter.cloudformation.yaml \
  --capabilities CAPABILITY_NAMED_IAM \
  --region us-east-2 \
  --parameter-overrides OpenIDConnectIdentityProvider=$(aws eks describe-cluster --name ${CLUSTER_NAME} | jq -r ".cluster.identity.oidc.issuer" | cut -c9-)
Note: AWS CLI version 2, the latest major version of the AWS CLI, is now stable and recommended for general use. For more information, see the AWS CLI version 2 installation instructions at: https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html

usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help
aws: error: argument --name: expected one argument

Waiting for changeset to be created..
Waiting for stack create/update to complete

Failed to create/update the stack. Run the following command
to fetch the list of events leading up to the failure
aws cloudformation describe-stack-events --stack-name Karpenter
```
It fails because I already have the node role created but check the usage error in the middle, I am not sure if it needs `--name` as a param.

Reply:
This set of instructions is for users setting up a first time. The role should be mastered by this stack, so try deleting the one you made manually and using the stack here.

This is an AWS specific YAML template, so it lives in the AWS docs. All of the /config is kubernetes and general for all providers.

The workflow should be (more or less)
1. Install Karpenter
2. Deploy CFN Stack
3. Patch ServiceAccount and restart for auth